Documentation/read/query.md

```diff
@@ -0,0 +1,175 @@
+must be a [query provider](../../Extending/Read/query_providers.md) for the return type.
+
+Dolittle has a very simple [repository for read models](read_model_repository.md) that can be used. You don't have to use it, as you decide entirely what you use internally - the query sits there as a contract and can also then be changed to accommodate a change in storage strategy.
+
+### Arguments
+
+With queries, sometimes you need to filter based on input. Any public property with
+a getter and a setter is considered an argument for the query.
+
+These can then be part of your query.
+
+```csharp
+using Dolittle.Read;
+
+public class EmployeesHiredAfter : IQueryFor<Employee>
+{
+    IReadModelRepositoryFor<Employee> _repository;
+
+    public EmployeesHiredAfter(IReadModelRepositoryFor<Employee> repository)
+    {
+        _repository = repository;
+    }
+
+    public DateTime HiredAfter { get; set; }
+
+    public IQueryable<Employee> Query => _repository.Query.Where(e => e.HiredDate >= HiredAfter);
+}
+```
+
+
+## Validation
+
+The arguments can have validators associated with them. All you need to do is to create something
+that inherits from `QueryValidationDescriptorFor<>` and point the generic argument to the
+query type.
+
+```csharp
+using Dolittle.Validation;
+using Dolittle.Read.Validation;
+
+public class EmployeeHiredAfterValidator : QueryValidationDescriptorFor<EmployeesHiredAfter>
+{
+    public EmployeeHiredAfterValidator()
+    {
+        // Require that hiring date is set after a certain date
+        ForArgument(e => e.HiredAfter).HasToBeGreaterThan(new DateTime(1985,1,1));
+    }
+}
+```
+
+## Paging
+
+Paging is not something you need to think about. This is what the [query provider](query_provider.md)
+deals with. Paging is a frontend concern and should not something you need to think about.
+In fact, paging is ideally a view concern, something decided as part of the interaction design.
+The client representations of queries has functionality related to paging.
+
+## Filtering
+
+Some features require filtering the read models on a model level. This could for instance be
+row level security with complex scenarios making it easier to do it as a post process after
+data has been received from the datasource.
+
+Dolittle has a global mechanism for filtering.
+
+```csharp
+using System.Collections.Generic;
+using System.Linq;
+using Dolittle.Read;
+
+public class MyReadModelFilter : ICanFilterReadModels
+{
+    public IEnumerable<IReadModel> Filter(IEnumerable<IReadModel> readModels)
+    {
+        var result = new List<IReadModel>();
+        foreach( var readModel in readModels )
+        {
+            if( readModel is Employee && ((Employee)readModel).HiredAfter > new DateTime(1985,1,1) )
+                result.Add(readModel);
+        }
+        return result;
+    }
```

Comment:
`return readModels
    .Where(rm => rm is Employee)
    .Where<Employee>(rm => rm.HiredAfter >= new DateTime(...));`